src/main/java/org/payw/npoem/config/security/JwtTokenProvider.java

```diff
@@ -0,0 +1,67 @@
+package org.payw.npoem.config.security;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Component
+public class JwtTokenProvider {
+
+    private String secretKey = "npoem";
+
+//    private long tokenValidTime = 30 * 60 * 1000L;
+    
+    private final UserDetailsService userDetailsService;
+
+    @PostConstruct
+    protected void init() {
+        secretKey = Base64.getEncoder().encodeToString(secretKey.getBytes());
+    }
+
+    public String createToken(String nickname) {
+        Claims claims = Jwts.claims().setSubject(nickname);
+        Date now = new Date();
+        return Jwts.builder()
+                .setClaims(claims)
+                .setIssuedAt(now)
+//                .setExpiration(new Date(now.getTime() + tokenValidTime))
```

Comment:
[question]
i think token require Expiration
Does token work without Expiration?

Reply:
Yes, It works well.